sktime/classification/distance_based/_time_series_neighbors.py

```diff
@@ -245,163 +222,40 @@ def kneighbors(self, X, n_neighbors=None, return_distance=True):
         """
         self.check_is_fitted()
 
-        if n_neighbors is None:
-            n_neighbors = self.n_neighbors
-        elif n_neighbors <= 0:
-            raise ValueError("Expected n_neighbors > 0. Got %d" % n_neighbors)
-        else:
-            if not np.issubdtype(type(n_neighbors), np.integer):
-                raise TypeError(
-                    "n_neighbors does not take %s value, "
-                    "enter integer value" % type(n_neighbors)
-                )
-
-        if X is not None:
-            query_is_train = False
-            X = check_array(X, accept_sparse="csr", allow_nd=True)
-        else:
-            query_is_train = True
-            X = self._fit_X
-            # Include an extra neighbor to account for the sample itself being
-            # returned, which is removed later
-            n_neighbors += 1
-
-        train_size = self._fit_X.shape[0]
-        if n_neighbors > train_size:
-            raise ValueError(
-                "Expected n_neighbors <= n_samples, "
-                " but n_samples = %d, n_neighbors = %d" % (train_size, n_neighbors)
-            )
-        n_samples = X.shape[0]
-        sample_range = np.arange(n_samples)[:, None]
-
-        n_jobs = effective_n_jobs(self.n_jobs)
-        if self._fit_method == "brute":
-
-            reduce_func = partial(
-                self._kneighbors_reduce_func,
-                n_neighbors=n_neighbors,
-                return_distance=return_distance,
-            )
-
-            # for efficiency, use squared euclidean distances
-            kwds = (
-                {"squared": True}
-                if self.effective_metric_ == "euclidean"
-                else self.effective_metric_params_
-            )
+        # self._X should be the stored _X
+        dist_mat = self._distance(X, self._X)
 
-            result = pairwise_distances_chunked(
-                X,
-                self._fit_X,
-                reduce_func=reduce_func,
-                metric=self.effective_metric_,
-                n_jobs=n_jobs,
-                **kwds,
-            )
-        else:
-            raise ValueError("internal: _fit_method not recognized")
+        neigh_ind = self.knn_estimator_.kneighbors(
```

Comment:
minor: the variable name `neigh_ind` could be confusing, as it could represent the pair `(neigh_dist, neigh_ind)`.  one possible alternative is `kneighbors_output`?

Reply:
clarified by using different name and adding comment

Reply:
resolved. happy with change